R/encoloc.R

```diff
@@ -113,20 +110,34 @@ coloc_wrapper <- function(xqtl_file, gwas_files,
                                 mutate(across(everything(), ~replace_na(., 0)))
 
     # Process xQTL data
-    xqtl_data <- readRDS(xqtl_file)[[xqtl_finemapping_obj]]
-    xqtl_lbf_matrix <- as.data.frame(xqtl_data$lbf_variable)
+    xqtl_data <- if (!is.null(xqtl_finemapping_obj)) readRDS(xqtl_file)[[1]][[xqtl_finemapping_obj]] else readRDS(xqtl_file)[[1]]
+    xqtl_lbf_matrix <- as.data.frame(xqtl_data$susie_result_trimmed$lbf_variable)
+    xqtl_lbf_matrix <- xqtl_lbf_matrix[xqtl_data$susie_result_trimmed$sets$cs_index,]
     colnames(xqtl_lbf_matrix) <- xqtl_data$variant_names
 
+
+    #add 'chr' in colnames 
+    add_chr_prefix <- function(df) {
+      colnames(df) <- ifelse(grepl("chr", colnames(df)), colnames(df), paste0("chr", colnames(df)))
+      return(df)
+    }
+
+    combined_gwas_lbf_matrix <- add_chr_prefix(combined_gwas_lbf_matrix)
+    xqtl_lbf_matrix <- add_chr_prefix(xqtl_lbf_matrix)
+    
+
     # Match column names and reorder matrices
     common_colnames <- intersect(colnames(xqtl_lbf_matrix), colnames(combined_gwas_lbf_matrix))
-    xqtl_lbf_matrix <- xqtl_lbf_matrix[, common_colnames, drop = FALSE]
-    combined_gwas_lbf_matrix <- combined_gwas_lbf_matrix[, common_colnames, drop = FALSE]
+    xqtl_lbf_matrix <- xqtl_lbf_matrix[, common_colnames, drop = FALSE] %>% as.matrix
+    combined_gwas_lbf_matrix <- combined_gwas_lbf_matrix[, common_colnames, drop = FALSE] %>% as.matrix
 
     # Report the number of dropped columns from xQTL matrix
     num_dropped_cols <- length(setdiff(colnames(xqtl_lbf_matrix), common_colnames))
     message("Number of columns dropped from xQTL matrix: ", num_dropped_cols)
 
     # Return the processed data for now
-    # FIXME: To complete this, we need to apply colocalization analysis on these single effect logBF and the enrichment analysis results computed by `xqtl_enrichment_wrapper`
-    return(list(xqtl_lbf_matrix = xqtl_lbf_matrix, combined_gwas_lbf_matrix = combined_gwas_lbf_matrix))
+    # FIXME: do we only keep trimmed lbf_matrix or not 
```

Comment:
we don't keey any lbf_matrix. I don't see where we would need it for any of the downsteeams analysis.